src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -230,87 +230,66 @@ public virtual void Execute()
                 timeout = (int)Test.Properties.Get(PropertyNames.Timeout);
 
 #if SILVERLIGHT || NETCF
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0)
+            if (Test is TestMethod)
                 RunTestOnOwnThread(timeout);
             else
                 RunTest();
 #elif PORTABLE
             RunTest();
 #else
-            ApartmentState currentApartment = Thread.CurrentThread.GetApartmentState();
+            currentApartment = Thread.CurrentThread.GetApartmentState();
 
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0 || currentApartment != TargetApartment && TargetApartment != ApartmentState.Unknown)
+            if (Test is TestMethod)
```

Comment:
Running a fixture in the STA simply means...

```
[TestFixture, Apartment(ApartmentState.STA)]
public class MyTestFixture
{
    ...
}
```

The fixture itself and all it's test cases are run in the STA. The cases do not actually need a separate thread, since the fixture is already in the STA.

A valid test would be a class like the above with a test case examining the current apartmentstate. We probably have such a test but it is only exercised using the dispatcher that is in effect when running. If we continue to support the SimpleWorkItemDispatcher, we probably should have separate tests for it somewhere. I didn't write them because i thought we would be getting rid of it much faster than we have.